app/controllers/chargeback_controller.rb

```diff
@@ -273,9 +300,136 @@ def cb_rate_form_field_changed
+      if !params[:metric].nil?
+        if params[:metric] != "null"
+          @edit[:selected_metrics] = params[:metric]
+        end
+        page << javascript_for_cb_button_add_metric_visibility(params[:metric] != "null")
+      elsif !params[:group].nil?
+        if @edit[:new][:level] == 'Compute' || @edit[:new][:level].nil?
+          rate_details_level = @sb[:rate_details].select { |k| k.rate == "0" && k.rate_type == "Compute" }
+        else
+          rate_details_level = @sb[:rate_details].select { |k| k.rate == "0" && k.rate_type == "Storage" }
+        end
+        @edit[:new][:metrics] = chargeback_details_metrics(rate_details_level, params[:group])
+        @edit[:new][:group] = params[:group]
+        page.replace_html("add_metric_fields", :partial => "cb_rate_add_metrics")
+      elsif !params[:level].nil?
+        if params[:level] == 'Compute'
+          rate_details_level = @sb[:rate_details].select { |k| k.rate == "0" && k.rate_type == "Compute" }
+        else
+          rate_details_level = @sb[:rate_details].select { |k| k.rate == "0" && k.rate_type == "Storage" }
+        end
+        @edit[:new][:groups] = chargeback_details_groups(rate_details_level)
+        @edit[:new][:level] = params[:level]
+        # showing all the metrics of the selected group
+        if @edit[:new][:groups].empty?
+          @edit[:new][:metrics] = []
+        else
+          @edit[:new][:metrics] = chargeback_details_metrics(rate_details_level, @edit[:new][:groups].first[1])
+        end
+        page.replace_html("add_metric_fields", :partial => "cb_rate_add_metrics")
+      end
+    end
+  end
+
+  def cb_add_rd
+    @edit = session[:edit]
+    # for save added rate details
+    @added_rds = []
+    case params[:button]
+    when "add_single"
+      selected_metrics = @edit[:selected_metrics].split(',')
+      selected_metrics.each do |sm|
+        # if the detail is not selectable by metric we select by description (for the Fixed rates datail)
+        rate_detail = @sb[:rate_details].select { |k| k.metric == sm || k.description == sm }
+        rate_detail[0].rate = 1.0
+        @added_rds.push(rate_detail[0])
+      end
+    when "add_all"
+      @added_rds = @sb[:rate_details].select { |k| k.rate == "0" }
+      @added_rds.each do |r|
+        r.rate = 1.0
+      end
+    end
+    cb_rate_set_form_vars
+    # flag to maintain the save button visible
+    @edit[:new][:added_metric] = true
+    replace_right_cell
+  end
+
+  def cb_remove_rd
+    @edit = session[:edit]
+    case params[:button]
+    when "remove_single"
+      rate_detail = @sb[:rate_details].select { |k| k.rate != "0" }[params['index'].to_i]
+      rate_detail.rate = "0"
+    when "remove_selected"
+      rate_details_temp = @sb[:rate_details].select { |k| k.rate != "0" }
+      @edit[:selected_rd_remove].each do |sm|
+        rate_details_temp[sm].rate = "0"
+      end
+      @sb[:rate_details] = (rate_details_temp + @sb[:rate_details]).uniq
+    end
```

Comment:
Same here, I see "0" and setting rate_details in `@sb` twice and think they're related changes... so maybe extract methods and see if you can keep the common logic in a common method.